prov/sm2/src/sm2_ep.c

```diff
@@ -722,6 +825,39 @@ static void sm2_free_entry(struct fi_peer_rx_entry *entry)
 	ofi_spin_unlock(&srx->lock);
 }
 
+int sm2_select_proto(void **desc, size_t iov_count, uint64_t op_flags,
+		     uint64_t total_len)
+{
+	struct ofi_mr *sm2_desc;
+	enum fi_hmem_iface iface;
+
+	sm2_desc = (struct ofi_mr *) *desc;
+	iface = sm2_desc->iface;
+
+	/* TODO: Move gdrcopy check out of non-cuda fast paths */
```

Comment:
Reminder, we fixed a bug on read path. We cannot use inject in that case.

https://github.com/ofiwg/libfabric/pull/8925

Reply:
we don't have RMA yet